src/drivers/display/nxp_elcdif/lv_nxp_elcdif.c

```diff
@@ -0,0 +1,93 @@
+/**
+ * @file lv_nxp_elcdif.c
+ *
+ * Driver for NXP's ELCD
+ */
+
+#include "lv_nxp_elcdif.h"
+
+#if LV_USE_NXP_ELCDIF == 1
+
+static lv_color_format_t lv_nxp_elcdif_to_lvgl_color_converter(elcdif_rgb_mode_config_t config)
+{
+    /*Handle color format conversion*/
+    lv_color_format_t color_format;
+
+    switch(config.pixelFormat) {
+        case kELCDIF_PixelFormatRAW8 :
+            color_format = LV_COLOR_FORMAT_L8;
+            break;
+        case kELCDIF_PixelFormatRGB565 :
+            color_format = LV_COLOR_FORMAT_RGB565;
+            break;
+        case kELCDIF_PixelFormatXRGB8888 :
+            color_format = LV_COLOR_FORMAT_XRGB8888;
+            break;
+        case kELCDIF_PixelFormatRGB888 :
+            color_format = LV_COLOR_FORMAT_RGB888;
+            break;
+        /*
+        There are some color formats in ELCDIF which LVGL does not support.
+        For these, use unknown format and drop a msg for the user
+        */
+        default :
+            color_format = LV_COLOR_FORMAT_UNKNOWN;
+            LV_LOG_WARN("Not supported color format in ELCDIF. Using LV_UNKNOWN!");
+    }
+    return color_format;
+}
+
+lv_display_t * lv_nxp_display_elcdif_create(elcdif_rgb_mode_config_t config, void * frame_buffer1,
```

Comment:
Only `lv_nxp_display_elcdif_create_direct` is required. The generic version is too generic. For partial we can create a dedicated `lv_nxp_display_elcdif_create_partial` function.